### PR TITLE
Fix not being able to dismiss new login toasts

### PR DESCRIPTION
**Pull request by turt2live**
`toastKey` is a thing.

Fixes https://github.com/vector-im/riot-web/issues/13923

### CODE REVIEW

File: src/stores/ToastStore.ts [hunk 1]
```diff
@@ -68,6 +68,11 @@ export default class ToastStore extends EventEmitter {
     }
 
     dismissToast(key) {
+        if (!this.toasts.some(t => t.key === key)) {
+            console.warn(`No toast for ${key} found - cannot dismiss. Check your toastKey`);
```
Comment (t3chguy) [inline src/stores/ToastStore.ts:72]: move into an else on line 88 plz, no point doing two array scans

Comment (t3chguy) [reply src/stores/ToastStore.ts:72]: also I think this warning will be SUPER noisy.

Comment (turt2live) [reply src/stores/ToastStore.ts:72]: why would it be noisy? Surely we shouldn't be hiding toasts that often...

Comment (t3chguy) [reply src/stores/ToastStore.ts:72]: every sync the mau toasts will try to hide
every update poll where there is no update the update toast will try to hide
etc

Comment (t3chguy) [reply src/stores/ToastStore.ts:72]: we don't keep state of whether a toast is open in a lot of places, so we don't have a falling edge to call hide on

Comment (t3chguy) [reply src/stores/ToastStore.ts:72]: noisy not as in often but as in very low signal to noise. Most of the times this warns will be very expected.

Comment (turt2live) [reply src/stores/ToastStore.ts:72]: oh fun. I'll just revert the commit then and we'll have to be careful.

**Pull request merged**